lib/Settings/BeforePreferenceSetEventListener.php

```diff
@@ -67,7 +67,7 @@ public function validatePreference(string $userId, string $key, string $value):
 		// "privacy" 0/1
 		if ($key === UserPreference::TYPING_PRIVACY
 			|| $key === UserPreference::READ_STATUS_PRIVACY) {
-			$valid = $value === (string)Participant::PRIVACY_PRIVATE || $value === (string)Participant::PRIVACY_PUBLIC;
+			$valid = (int)$value === Participant::PRIVACY_PRIVATE || (int)$value === Participant::PRIVACY_PUBLIC;
```

Comment:
![grafik](https://github.com/user-attachments/assets/b147978e-f237-495d-aecb-f8dcf2d66b07)

:shrug: we can not cast the value on validation if we afterwards continue with the provided value.

But yeah we should add a test instead :P

Reply:
Maybe `(string)$value === (string)Participant::PRIVACY_PRIVATE` would be better here?

Currently it's breaking only main and 30.1, but I don't want to adjust frontend for that; otherwise Talk desktop won't work with previous versions. So if anything helps to keep previous payload valid, I'm up for it